pinot-core/src/main/java/org/apache/pinot/core/plan/GroupByPlanNode.java

```diff
@@ -50,10 +54,35 @@ public GroupByPlanNode(IndexSegment indexSegment, QueryContext queryContext) {
   }
 
   @Override
-  public GroupByOperator run() {
+  public Operator<GroupByResultsBlock> run() {
     assert _queryContext.getAggregationFunctions() != null;
     assert _queryContext.getGroupByExpressions() != null;
 
+    if (_queryContext.hasFilteredAggregations()) {
+      return filteredGroupByPlan();
+    }
+
+    return nonFilteredGroupByPlan();
+  }
+
+  private FilteredGroupByOperator filteredGroupByPlan() {
```

Comment:
(minor) Rename to `buildFilteredGroupByPlan()` and `buildNonFilteredGroupByPlan()`